### ISSUE TITLE
Ignore all Steam messages coming from people who are not on our friends list

### ISSUE THREAD

**Issue by scholtzm** [Jessecar96/SteamBot#702]
This has been bothering me for a while now.

SteamBot *does not* ignore [Steam messages](https://github.com/Jessecar96/SteamBot/blob/master/SteamBot/Bot.cs#L359) coming from people who are not on our friends list. This way, you can even do a trade. Regular Steam client just ignores these messages.

We already have [this](https://github.com/Jessecar96/SteamBot/blob/master/SteamBot/Bot.cs#L359) so the fix should be fairly straight forward.

cc @BlueRaja @WildCard65 @FrozenHaxor

**Comment by Bottswana** [comment]
Hmm. Can you actually initiate a trade with a bot not on your friends list, though?

**Comment by scholtzm** [comment]
Yes you can.

**Comment by Bottswana** [comment]
That’s a pain. But as you say, it should just be a cause of altering the message handler to not pass steam chats or trade requests on from non friends.

I’m surprised there’s not some form of check on the steam network itself for trading with non friends. Hmm.

**Comment by FrozenHaxor** [comment]
Also somehow blocked people count as friends. For example if you block someone and remove them, [this](https://github.com/SteamRE/SteamKit/blob/6b86fdbb280680879d2300af1d6fe53895e3a97d/SteamKit2/SteamKit2/Steam/Handlers/SteamFriends/SteamFriends.cs#L87) won't decrease, which is extremely annoying. I assume this is just another example of the stated above... the wonders of Steam.

**Comment by WildCard65** [comment]
@FrozenHaxor I think that issue would be more useful if reported to the SteamKit repo

**Comment by scholtzm** [comment]
That's actually not an issue, but that's how Steam works. People on your block list are part of your "global" friends list. You need to check relationship between you and the person. Another example is someone who adds you - this person automatically becomes part of your friends list with `EFriendRelationship.RequestRecipient`.

**Comment by Bottswana** [comment]
Yeah I wouldn't say this is SteamKit's fault. It's just doing it's job passing on the messages it receives from the network; I mean, if it's anyones fault it's the steam network itself.

**Comment by BlueRaja** [comment]
> You need to check relationship between you and the person.
I wrote that code, I wasn't aware of this.  How do we 'check relationship'?

**Comment by doktokto** [comment]
Wow, I don't ever thinked that it's bug, I just used 
```
if (Bot.SteamFriends.GetFriendRelationship(OtherSID) == EFriendRelationship.Friend)
{
...
}
```

**Comment by scholtzm** [comment]
@BlueRaja Which code is "that code"? :smile: You can check the relationship the way @doktokto described.

**Comment by Dynsay** [comment]
I dunno everywhere bugs, i tested this bot and workin fine. I wrote strange, key and scrap userhandlers after March 9th, workin all wo/ bugs, errors. In my case i using my bots.
